kickstarter/android-oss
# NTV-315: Migrate ComparatorUtils.java to kotlin FloatExt

# 📲 What

Remove `ComparatorUtils.java` and migrate to kotlin ext. Kotlin has built in `compareTo` float method, but because these methods are being used in java classes that have not yet been deprecated, we need an extension function to make this available to java. 

# 🤔 Why

Ongoing effort to migrate to kotlin. 

# 🛠 How

Extension function just calls kotlins built in `compareTo` function, but in descending order (the numbers are flipped).

# 👀 See
These are the two sections effected by the change on the creator dashboard:
![Screenshot_1641328368](https://user-images.githubusercontent.com/19390326/148250552-af379845-e0a9-4f87-bf5d-3acf3e723fb7.png)

# 📋 QA

- Login to a creator account with a project with multiple rewards
- Tap the menu button and select dashboard
- Select a project with multiple rewards
- Make sure the two sections in the above screenshot ("Top Rewards" and "Top Pledge Sources"), the list contents are in descending order

# Story 📖

[NTV-315: Migrate to Kotlin Extension ComparatorUtils.java](https://kickstarter.atlassian.net/browse/NTV-315)
